server/routes/createGoal/createGoalController.ts

```diff
@@ -22,9 +22,13 @@ export default class CreateGoalController {
     req.session.newGoals = req.session.newGoals || []
 
     if (isEditMode(req)) {
+      if (!req.session.newGoals[parseInt(goalIndex, 10) - 1]) {
```

Comment:
This handles both undefined and NaN, we could have redirected the user to the start of the create goal flow instead of a 404 however, its mostly likely they're changing the number manually (which they shouldn't be doing) but also if it is a genuine error we can see the 404 in the logs to fix it.

Reply:
100% agree; nice simple solution. Nice 👍